app.js

```diff
@@ -3,26 +3,68 @@
 const express = require('express');
 const app = express();
 const port = process.env.PORT || 3000;
-const playersData = require('');//should require the data.json file
+const playersData = require('./data.json');//should require the data.json file
 app.use(express.json());
 
 app.get('/players', (req, res) => {
     //should respond with the "players" array inside playersData and Status 200    
+    try {
+        res.status(200).json({
+          data: playersData.players
+        });
+      } catch (err) {
+        res.status(400).json({
+          message: "Error",
+          err
+        });
+      }
 });
 
 app.get('/players/:role', (req, res) => {
     //should respond with only the players that have with the especified role. Status 200.
     //If there's no player with the specified role it should respond with {"error": "No player found"} and Status 404.
+    let { role } = req.params;    
+    try {
+        let playersrole = playersData.players.filter(player => player.role===role);
+        if(playersrole!==[])
```

Comment:
You should probably use the .length property to check if an array is empty. This comparison returns true because is comparing the references of the array objects.

You can read more about this in https://www.freecodecamp.org/news/check-if-javascript-array-is-empty-or-not-with-length/

And here:
https://www.py4u.net/discuss/289209